Source/Generic/ECTestComparisons.m

```diff
@@ -13,7 +13,13 @@ - (BOOL)matches:(id)item2 options:(ECTestComparisonOptions)options block:(ECTest
 	Class c1 = [self class];
 	Class c2 = [item2 class];
 	NSString* context = nil;
-	if (c1 == c2)
+	
+	//macOS 10.12 Sierra introduces __NSSingleEntryDictionaryI for single-entry dicts and similar for NSArray
```

Comment:
A cleaner way to fix this might be do just do `[[self nameForMatching] isEqual:[item2 nameForMatching]]`.

That method is already there to ensure that we generate the same name for class clusters. It was originally so that the text dumps/diffs didn't vary, but we can re-use it here.